include/boost/geometry/algorithms/detail/envelope/interface.hpp

```diff
@@ -57,13 +62,15 @@ struct envelope
                              Box& box,
```

Comment:
why do not split here in two structs and dispatch as in area and distance?

Reply:
Because here it's sufficient to distinguish between strategies on function level, the compiler can pick the correct function automatically. With area and distance the function return type was based on strategy and this caused problems. Even the `Geometry` parameter could probably be moved.